src/mmrelay/matrix_utils.py

```
@@ -2137,43 +2536,68 @@ async def login_matrix_bot(
+            if credentials_path:
+                path_exists = await asyncio.to_thread(os.path.exists, credentials_path)
+            else:
+                path_exists = False
+            existing_credentials_path = credentials_path if path_exists else None
+            if existing_credentials_path is not None:
+
+                def _load_direct(path: str) -> dict[str, Any] | None:
+                    try:
+                        with open(path, "r", encoding="utf-8") as f:
+                            return cast(dict[str, Any], json.load(f))
+                    except Exception:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)